src/zope/testrunner/tests/testrunner-subunit.rst

```diff
@@ -0,0 +1,721 @@
+Subunit Output
+==============
+
+Subunit is a streaming protocol for interchanging test results. More
+information can be found at https://launchpad.net/subunit.  To enable
+support for ``subunit``, install it and its dependencies via the
+``zope.testrunner`` extra:
+
+    $ pip install zope.testrunner[subunit]
+
+or directly:
+
+    $ pip install python-subunit
+
+First we need to make a temporary copy of the entire testing directory:
+
+    >>> import os.path, sys, tempfile, shutil
+    >>> tmpdir = tempfile.mkdtemp()
+    >>> directory_with_tests = os.path.join(tmpdir, 'testrunner-ex')
+    >>> source = os.path.join(this_directory, 'testrunner-ex')
+    >>> n = len(source) + 1
+    >>> for root, dirs, files in os.walk(source):
+    ...     dirs[:] = [d for d in dirs if d != ".svn"] # prune cruft
+    ...     os.mkdir(os.path.join(directory_with_tests, root[n:]))
+    ...     for f in files:
+    ...         _ = shutil.copy(os.path.join(root, f),
+    ...                         os.path.join(directory_with_tests, root[n:], f))
+
+    >>> defaults = [
+    ...     '--path', directory_with_tests,
+    ...     '--tests-pattern', '^sampletestsf?$',
+    ...     ]
+
+    >>> from zope import testrunner
+
+
+Basic output
+------------
+
+Subunit output is line-based, with a 'test:' line for the start of each test
+and a 'successful:' line for each successful test.
+
+Zope layer set up and tear down events are represented as tests tagged with
```

Comment:
I think it's "setup and teardown events" when "set up" and "tear down" are not verbs, but I'm not a native English speaker.